pkg/apis/core/types.go

```diff
@@ -5653,6 +5660,29 @@ type TopologySpreadConstraint struct {
 	// in their corresponding topology domain.
 	// +optional
 	LabelSelector *metav1.LabelSelector
+	// MinDomains indicates a minimum number of eligible domains.
+	// When the number of eligible domains with matching topology keys is less than `minDomains`,
+	// Pod Topology Spread treats "global minimum" as 0.
+	// As a result, when the number of eligible domains is less than `minDomains`,
+	// scheduler won't schedule more than maxSkew Pods to those domains.
+	// The default value is 1. Valid values are integers greater than 0.
+	// When value is different from 1, WhenUnsatisfiable must be DoNotSchedule.
+	//
+	// For example, in a 3-zone cluster, MaxSkew is set to 2, MinDomains is set to 5 and pods with the same
+	// labelSelector spread as 2/2/2:
```

Comment:
I was wondering what will happen if we set `MaxSkew` here to 10.

Reply:
This is described in the above part. (L5666)
> 	// As a result, when the number of eligible domains is less than `minDomains`,
	// scheduler won't schedule more than maxSkew Pods to those domains.

---

Even if we set `MaxSkew=10` (and the other parameters and situation are the same as this example), global minimum is treated as 0. 
But, before spreading pods as 10/10/10, scheduling will not fail as the example.
In Filter, we judged the Nodes with the following criteria, 
```
('existing matching num' + 'if self-match (1 or 0)' - 'global min matching num') <= 'maxSkew'
```
and, before spreading pods as 10/10/10, `x + 1 - 0 <= 10` will be `true` in at least one Node .

Do you think that the current documentation is not enough? If this documentation is difficult to understand for someone who understands schedulers (includes you), then many users also may not be able to understand this.

Reply:
So, even we have less eligible domains than `MinDomains`, we can still schedule pods until we reach the Skew limit, right? I think this was the point I'm confusing. I thought `MinDomains` was a hard constraint. I wonder whether this will also be mis-understood by end-users.

Reply:
Yes, your understanding is right.
Small change, but I added the statement that skew is calculated after MinDomains are taken into account in https://github.com/kubernetes/kubernetes/pull/107674/commits/03a6294fdf9fb4a5e1f2c7898969b9c0eb8fdf0d.

Reply:
@kerthcet this was discussed in the original KEP, and had a consensus that the global minimum is read as 0 (like a virtual topology not existing yet) if the total number of topologies is less than minDomains.